Adding workaround fix for syslog issue found in test_sequential_restart.py

kbabujp opened the pull request:
### Description of PR

Adding workaround fix for syslog issue found in test_sequential_restart.py

Summary:

In 202111 PTF T1 topo, after running platform_tests/test_sequential_restart.py with all BGP session established and routes populated,  log messages are not getting written in to /var/log/syslog. The log messages are getting stuck. When we execute “systemctl stop syslog”, the stuck messages are getting written in to /var/log/syslog. This issue happens only when we had BGP session up and all the neighbors in T1 setup being up.

Adding "sudo systemctl restart syslog" at the end of test to fix the issue.

Fixes # 6995

### Type of change

- [x] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)


### Back port request
- [ ] 201911
- [ ] 202012
- [ ] 202205

### Approach
#### What is the motivation for this PR?
To fix the issue reported in 6995
#### How did you do it?

#### How did you verify/test it?
Run the test and reported issue is not seen

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation

prgeor commented:
@kbabujp can you point me to the link for the issue Fixes #6995

prgeor commented:
@kbabujp please attach failure log of the test_sequential_restart.py

prgeor commented:
@kbabujp this seems to be a workaround but does not address the actual issue, which is syslog messages are not coming after service restarts.

kbabujp commented:
@prgeor , Thanks for looking into this PR. Test platform_tests/test_sequential_restart.py::test_restart_swss[None] passes but after completing this test, no syslog messages seen.

This is reproduced manually with out running the script in https://github.com/sonic-net/sonic-mgmt/issues/6995.
I have attached the other details requested there.

If we run any PTF test that verify syslog messages after platform_tests/test_sequential_restart.py, they may error or fail as the loganalyser cannot add/verify start/stop marker in syslog.

kbabujp commented:
@prgeor , Please find the attached test run logs.
[platform-seq-restart.log](https://github.com/sonic-net/sonic-mgmt/files/10431858/platform-seq-restart.log)

prgeor commented:
@kbabujp how is config reload fixing the issue? any reason? can you restart syslogd to see if that fixes?

kbabujp commented:
@prgeor , Restarting syslog (systemctl restart syslog) instead of config reload also fix the issue. 
But just to be on safer side and avoid other problems, using the config reload.

prgeor commented:
> @prgeor , Restarting syslog (systemctl restart syslog) instead of config reload also fix the issue. But just to be on safer side and avoid other problems, using the config reload.

@kbabujp what do you mean by to be on safer side? what other problems?

kbabujp commented:
> > @prgeor , Restarting syslog (systemctl restart syslog) instead of config reload also fix the issue. But just to be on safer side and avoid other problems, using the config reload.
> 
> @kbabujp what do you mean by to be on safer side? what other problems?

@prgeor , there is no other issue. I thought config reload is better compare to restarting syslog. we can do "systemctl restart syslog" to overcome this issue.

prgeor commented:
> > > @prgeor , Restarting syslog (systemctl restart syslog) instead of config reload also fix the issue. But just to be on safer side and avoid other problems, using the config reload.
> > 
> > 
> > @kbabujp what do you mean by to be on safer side? what other problems?
> 
> @prgeor , there is no other issue. I thought config reload is better compare to restarting syslog. we can do "systemctl restart syslog" to overcome this issue.

@kbabujp just restarting syslog service would be better instead of restart all services with "config reload"

kbabujp commented:
> > > > @prgeor , Restarting syslog (systemctl restart syslog) instead of config reload also fix the issue. But just to be on safer side and avoid other problems, using the config reload.
> > > 
> > > 
> > > @kbabujp what do you mean by to be on safer side? what other problems?
> > 
> > 
> > @prgeor , there is no other issue. I thought config reload is better compare to restarting syslog. we can do "systemctl restart syslog" to overcome this issue.
> 
> @kbabujp just restarting syslog service would be better instead of restart all services with "config reload"

@prgeor , I have removed config reload and added sylog restart and pushed the change to this branch.

kbabujp commented:
Hi, The test failures are not related to this change. Could you please re-run the tests.

wangxin commented:
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/platform_tests/test_sequential_restart.py:11:1: F401 'tests.common.config_reload' imported but unused<br/>tests/platform_tests/test_sequential_restart.py:12:1: F401 'tests.common.fixtures.conn_graph_facts.conn_graph_facts' imported but unused<br/>tests/platform_tests/test_sequential_restart.py:39:1: E302 expected 2 blank lines, found 1<br/>tests/platform_tests/test_sequential_restart.py:51:1: E302 expected 2 blank lines, found 1<br/>tests/platform_tests/test_sequential_restart.py:55:55: E225 missing whitespace around operator<br/>tests/platform_tests/test_sequential_restart.py:71:121: E501 line too long (129 > 120 characters)<br/>tests/platform_tests/test_sequential_restart.py:92:93: F811 redefinition of unused 'conn_graph_facts' from line 12<br/>tests/platform_tests/test_sequential_restart.py:92:121: E501 line too long (126 > 120 characters)<br/>tests/platform_tests/test_sequential_restart.py:104:27: E231 missing whitespace after ':'<br/>tests/platform_tests/test_sequential_restart.py:111:94: F811 redefinition of unused 'conn_graph_facts' from line 12<br/>tests/platform_tests/test_sequential_restart.py:111:121: E501 line too long (127 > 120 characters)<br/>...<br/>[truncated extra lines, please run pre-commit locally to view full check results]<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

wangxin commented:
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/platform_tests/test_sequential_restart.py:11:1: F401 'tests.common.config_reload' imported but unused<br/>tests/platform_tests/test_sequential_restart.py:12:1: F401 'tests.common.fixtures.conn_graph_facts.conn_graph_facts' imported but unused<br/>tests/platform_tests/test_sequential_restart.py:57:55: E225 missing whitespace around operator<br/>tests/platform_tests/test_sequential_restart.py:73:121: E501 line too long (129 > 120 characters)<br/>tests/platform_tests/test_sequential_restart.py:94:93: F811 redefinition of unused 'conn_graph_facts' from line 12<br/>tests/platform_tests/test_sequential_restart.py:94:121: E501 line too long (126 > 120 characters)<br/>tests/platform_tests/test_sequential_restart.py:106:27: E231 missing whitespace after ':'<br/>tests/platform_tests/test_sequential_restart.py:113:94: F811 redefinition of unused 'conn_graph_facts' from line 12<br/>tests/platform_tests/test_sequential_restart.py:113:121: E501 line too long (127 > 120 characters)<br/>tests/platform_tests/test_sequential_restart.py:118:121: E501 line too long (151 > 120 characters)<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

wangxin commented:
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Failed<br/>- hook id: trailing-whitespace<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/platform_tests/test_sequential_restart.py<br/><br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/platform_tests/test_sequential_restart.py:106:27: E231 missing whitespace after ':'<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

wangxin commented:
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Failed<br/>- hook id: trailing-whitespace<br/>- exit code: 1<br/>- files were modified by this hook<br/><br/>Fixing tests/platform_tests/test_sequential_restart.py<br/><br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Passed<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```

wangxin commented:
@kbabujp Can you also update the PR description to comply with the actual change of this PR?
From the code, I see that `systemctl restart syslog` is performed to workaround the issue. PR description says that `config_reload` is performed. They do not match.

kbabujp commented:
> @kbabujp Can you also update the PR description to comply with the actual change of this PR? From the code, I see that `systemctl restart syslog` is performed to workaround the issue. PR description says that `config_reload` is performed. They do not match.

@wangxin , I have updated the PR description. Please check.

kbabujp commented:
/azp run

wangxin commented on tests/platform_tests/test_sequential_restart.py:
```diff
@@ -8,8 +8,6 @@
 import pytest
 
-from tests.common import config_reload
-from tests.common.fixtures.conn_graph_facts import conn_graph_facts
```
Can you address this comment?

kbabujp replied:
Thanks for checking this. I have addressed the comment. I have added "# noqa F811" also for another failure.

wangxin reviewed (changes requested):
I have a comment about fixture importing. Can you check?

wangxin commented:
The [pre-commit](http://pre-commit.com/) check detected issues in the files touched by this pull request.
The detected issues may be old or new. For new issues, please try to fix them.

For old issues, it is not mandatory to fix them because they were not caused by this change. It is unfair to blame
author of this pull request. But if you can take extra effort to fix the old issues as well, that would be great!

Detailed pre-commit check results:
<samp>trim trailing whitespace.................................................Passed<br/>fix end of files.........................................................Passed<br/>check yaml...........................................(no files to check)Skipped<br/>check for added large files..............................................Passed<br/>check python ast.........................................................Passed<br/>flake8...................................................................Failed<br/>- hook id: flake8<br/>- exit code: 1<br/><br/>tests/platform_tests/test_sequential_restart.py:96:34: F811 redefinition of unused 'conn_graph_facts' from line 12<br/>tests/platform_tests/test_sequential_restart.py:116:35: F811 redefinition of unused 'conn_graph_facts' from line 12<br/></samp>

To run the pre-commit checks locally, you can follow below steps:
1. Ensure that default python is python3. In sonic-mgmt docker container, default python is python2. You can run
   the check by activating the python3 virtual environment in sonic-mgmt docker container or outside of sonic-mgmt
   docker container.
2. Ensure that the `pre-commit` package is installed:
```
sudo pip install pre-commit
```
3. Go to repository root folder
4. Install the pre-commit hooks:
```
pre-commit install
```
5. Use pre-commit to check staged file:
```
pre-commit
```
6. Alternatively, you can check committed files using:
```
pre-commit run --from-ref <commit_id> --to-ref <commit_id>
```
